src/Analyzer.Utilities/Extensions/ISymbolExtensions.cs

```diff
@@ -107,6 +107,68 @@ public static SymbolVisibility GetResultantVisibility(this ISymbol symbol)
 
             return visibility;
         }
+
+        public static bool MatchMemberDerived(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.ContainingType.DerivesFrom(type) && member.MetadataName == name;
+        }
+
+        public static bool MatchMethodDerived(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Method && member.MatchMemberDerived(type, name);
+        }
+
+        public static bool MatchMethod(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Method && member.MatchMember(type, name);
+        }
+
+        public static bool MatchPropertyDerived(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Property && member.MatchMemberDerived(type, name);
+        }
+
+        public static bool MatchFieldDerived(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Field && member.MatchMemberDerived(type, name);
+        }
+
+        public static bool MatchMember(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.ContainingType == type && member.MetadataName == name;
+        }
+
+        public static bool MatchProperty(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Property && member.MatchMember(type, name);
+        }
+
+        public static bool MatchField(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Field && member.MatchMember(type, name);
+        }
+
+        public static ITypeSymbol GetVariableSymbolType(this ISymbol symbol)
+        {
+            if (symbol == null)
+            {
+                return null;
+            }
+            SymbolKind kind = symbol.Kind;
+            switch (kind)
+            {
+                case SymbolKind.Field:
+                    return ((IFieldSymbol)symbol).Type;
+                case SymbolKind.Local:
+                    return ((ILocalSymbol)symbol).Type;
+                case SymbolKind.Parameter:
+                    return ((IParameterSymbol)symbol).Type;
+                case SymbolKind.Property:
+                    return ((IPropertySymbol)symbol).Type;
+                default:
```

Comment:
This method is not used, please remove. If it is used by other 2 XML analyzers, then please move it to Desktop analyzer's helper class instead and consider renaming the method (field and property are not variable.)